infrastructures/openshift/src/main/java/org/eclipse/che/workspace/infrastructure/openshift/provision/AsyncStorageProvisioner.java

```diff
@@ -0,0 +1,384 @@
+  /**
+   * The authorized_keys file in SSH specifies the SSH keys that can be used for logging into the
+   * user account for which the file is configured.
+   */
+  private static final String AUTHORIZED_KEYS = "authorized_keys";
+  /**
+   * Name of the asynchronous storage Pod and Service. Rsync command will use this Service name for
+   * communications: e.g.: rsync ${RSYNC_OPTIONS} --rsh="ssh ${SSH_OPTIONS}" async-storage:/{PATH}
+   */
+  static final String ASYNC_STORAGE = "async-storage";
+  /** The name suffix for ConfigMap with SSH configuration */
+  static final String ASYNC_STORAGE_CONFIG = "async-storage-config";
+  /** The path of mount storage volume for file persist */
+  private static final String ASYNC_STORAGE_DATA_PATH = "/" + ASYNC_STORAGE;
+  /** The path to the authorized_keys */
+  private static final String SSH_KEY_PATH = "/.ssh/" + AUTHORIZED_KEYS;
+  /** The name of SSH key pair for rsync */
+  static final String SSH_KEY_NAME = "rsync-via-ssh";
+  /** The name of volume for mounting configuration map and authorized_keys */
+  private static final String CONFIG_MAP_VOLUME_NAME = "async-storage-configvolume";
+  /** */
+  private static final String STORAGE_VOLUME = "async-storage-data";
+
+  private static final Logger LOG = LoggerFactory.getLogger(AsyncStorageProvisioner.class);
+
+  private final String sidecarImagePullPolicy;
+  private final String pvcQuantity;
+  private final String asyncStorageImage;
+  private final String pvcAccessMode;
+  private final String pvcStrategy;
+  private final String pvcName;
+  private final String pvcStorageClassName;
+  private final SshManager sshManager;
+  private final OpenShiftClientFactory clientFactory;
+
+  @Inject
+  public AsyncStorageProvisioner(
+      @Named("che.workspace.sidecar.image_pull_policy") String sidecarImagePullPolicy,
+      @Named("che.infra.kubernetes.pvc.quantity") String pvcQuantity,
+      @Named("che.infra.kubernetes.async.storage.image") String asyncStorageImage,
+      @Named("che.infra.kubernetes.pvc.access_mode") String pvcAccessMode,
+      @Named("che.infra.kubernetes.pvc.strategy") String pvcStrategy,
+      @Named("che.infra.kubernetes.pvc.name") String pvcName,
+      @Named("che.infra.kubernetes.pvc.storage_class_name") String pvcStorageClassName,
+      SshManager sshManager,
+      OpenShiftClientFactory openShiftClientFactory) {
+    this.sidecarImagePullPolicy = sidecarImagePullPolicy;
+    this.pvcQuantity = pvcQuantity;
+    this.asyncStorageImage = asyncStorageImage;
+    this.pvcAccessMode = pvcAccessMode;
+    this.pvcStrategy = pvcStrategy;
+    this.pvcName = pvcName;
+    this.pvcStorageClassName = pvcStorageClassName;
+    this.sshManager = sshManager;
+    this.clientFactory = openShiftClientFactory;
+  }
+
+  public void provision(OpenShiftEnvironment osEnv, RuntimeIdentity identity)
+      throws InfrastructureException {
+    if (!parseBoolean(osEnv.getAttributes().get(ASYNC_PERSIST_ATTRIBUTE))) {
```

Comment:
NPE if the attributes don't contain the async attribute.

Reply:
```
public static boolean parseBoolean(String s) {
        return "true".equalsIgnoreCase(s);
    }
```

```
public boolean equalsIgnoreCase(String anotherString) {
        return (this == anotherString) ? true
                : (anotherString != null)
                && (anotherString.length() == length())
                && regionMatches(true, 0, anotherString, 0, length());
    }
```